modules/README.md

```diff
@@ -115,35 +119,235 @@ Kudos to @mfw78
      */
     function isValidSignature(
         address account,
         address sender,
-        bytes32 _hash,
+        bytes32 messageHash,
         bytes32 domainSeparator,
-        bytes32 typeHash,
-        bytes calldata encodeData,
-        bytes calldata payload
+        bytes32 hashStruct
+        bytes calldata signatures
     ) external view returns (bytes4 magic);
 }
 ```
 
+### Hooks for signature validator
+
+Signature validation is a critical part of smart contract accounts. A flawed implementation of validation logic can lead to approving a transaction with an invalid signature. Signature validator hooks can be used to add safety net to the signature validation process and hold the power to block a signature validator from approving unauthorized actions. Hooks can be enabled for all validators before and after execution of the validation function. This can be done by adding contract implementing `ISignatureValidatorHook` interface to the `SignatureValidatorManager` by the account. The hooks are not specific to a domain and are executed for all signature validations. The hooks function(s) should revert on failed checks.
+
+```solidity
+interface ISignatureValidatorHooks {
+    /**
+     * @param account Address of the account for which signature is being validated
+     * @param validator Address of the validator contract to be used for signature validation. This address will be account address in case of default signature validation flow is used.
+     * @param payload The payload provided for the validation
+     * @return result bytes containing the result
+     */
+    function preValidationHook(address account, address validator, bytes calldata payload) external returns (bytes memory result);
+
+    /**
+     * @param account Address of the account for which signature is being validated
+     * @param preValidationData Data returned by preValidationHook
+     * @return result bytes containing the result
+     */
+    function postValidationHook(address account, bytes calldata preValidationData) external returns (bytes memory result);
+}
+```
+
+### Signature validator manager
+
+A signature validator manager must implement the following interface.
+
+```solidity
+interface ISafeProtocolSignatureValidatorManager {
+
+    /**
+     * @notice A view function that the Manager will call when an account has enabled this contract as a function handler in the Manager for function isSignatureValid(bytes32,bytes)
+     * @param account Address of the account whose signature validator is to be used
+     * @param sender Address requesting signature validation
+     * @param data Calldata containing the 4 bytes function selector, 4 bytes signature selector, and one of the below:
+     *              encoded bytes containing bytes32 domain separator, bytes32 messageHash and bytes32 signatures
+     *              OR
+     *              encoded bytes containing bytes32 messageHash and bytes signatures
+     */
+    function handle(
```

Comment:
As mentioned above, I would remove this method from the interface definition (and from this PR)